lib/scenic/statements.rb

```diff
@@ -110,13 +109,13 @@ def replace_view(name, version: nil, revert_to_version: nil, materialized: false
         raise ArgumentError, "version is required"
       end
 
-      if materialized
-        raise ArgumentError, "Cannot replace materialized views"
-      end
-
       sql_definition = definition(name, version)
 
-      Scenic.database.replace_view(name, sql_definition)
+      if materialized
+        Scenic.database.replace_materialized_view(name, sql_definition)
```

Comment:
If you want to re-apply or attempt to re-apply some of the previous  existing indexes you should use `#update_view` instead.